### PR TITLE
Add reading support nwb x files

### PR DESCRIPTION
**Pull request by t-b**
This is the child class for nwb_reader.

To use it:

```python
In [1]: import pynwb

In [2]: import ipfx.reader
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-2-e01e64f286bc> in <module>()
----> 1 import ipfx.reader

ImportError: No module named reader

In [3]: import ipfx.nwb_reader

In [4]: reader = ipfx.nwb_reader.create_nwb_reader("tests/reference_abf_nwb/2018_03_20_0005.nwb")

In [5]: reader.get_sweep_names()

In [10]: reader.get_sweep_data(0)
Out[10]:
{'index_range': (0, 899999),
 'response': array([-0.06601563, -0.0659668 , -0.06594849, ..., -0.06213989,
        -0.06210938, -0.06208497], dtype=float32),
 'sampling_rate': 50000.0,
 'stimulus': array([ 0.,  0.,  0., ...,  0.,  0.,  0.], dtype=float32),
 'stimulus_unit': 'Amps'}
```

### CODE REVIEW

**Comment by t-b** [comment]
For getting the sample files execute ` py.test --do-x-nwb-tests test_x_nwb.py` with python 3.6.

File: ipfx/nwb_reader.py [hunk 1]
```diff
@@ -3,6 +3,15 @@
 import ipfx.stim_features as st
 
 
+def get_long_unit_name(unit):
```
Comment (sgratiy) [inline ipfx/nwb_reader.py]: This function interprets the nwb data and is better suited for the nwb_reader class. The stim_feaures module includes functions for determining features from traces and should hopefully be independent of any nwb functionality

Comment (dyf) [reply ipfx/nwb_reader.py]: Good point @sgratiy .  @t-b could you address this in the next PR?

File: ipfx/nwb_reader.py [hunk 2]
```diff
@@ -207,16 +343,19 @@ class NwbMiesReader(NwbReader):
...
+            sweep_response = f[self.acquisition_path]["data_%05d_AD0" %
+                                                      sweep_number]
             response = sweep_response["data"].value
             hz = 1.0 * sweep_response["starting_time"].attrs['rate']
-            sweep_stimulus = f['stimulus']['presentation']["data_%05d_DA0" % sweep_number]
+            sweep_stimulus = f[self.stimulus_path]["data_%05d_DA0" %
+                                                   sweep_number]
             stimulus = sweep_stimulus["data"].value
 
             if 'unit' in sweep_stimulus["data"].attrs:
```
Comment (sgratiy) [inline ipfx/nwb_reader.py]: Could use get_long_unit_name here as well

File: ipfx/nwb_reader.py (hunk 1 above)
Comment (t-b) [reply ipfx/nwb_reader.py]: @dyf @sgratiy I can certainly make this a NwbReader class function. I'll prepare a PR.

File: ipfx/nwb_reader.py (hunk 2 above)
Comment (t-b) [reply ipfx/nwb_reader.py]: @sgratiy I've seen that you have already done that in your PR. If you want me to include the changes in my PR as well just ping me.

**Pull request merged**